Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/CertificatePersonIdentifier.kt

```diff
@@ -1,26 +1,65 @@
 package de.rki.coronawarnapp.covidcertificate.common.certificate
 
-import com.google.gson.annotations.SerializedName
+import androidx.annotation.VisibleForTesting
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
 import de.rki.coronawarnapp.covidcertificate.common.exception.InvalidHealthCertificateException.ErrorCode.DOB_MISMATCH
 import de.rki.coronawarnapp.covidcertificate.common.exception.InvalidHealthCertificateException.ErrorCode.NAME_MISMATCH
 import de.rki.coronawarnapp.covidcertificate.common.exception.InvalidVaccinationCertificateException
 import de.rki.coronawarnapp.util.HashExtensions.toSHA256
+import de.rki.coronawarnapp.util.list.hasIntersect
 import timber.log.Timber
 
 data class CertificatePersonIdentifier(
-    @SerializedName("dateOfBirth") val dateOfBirthFormatted: String,
-    @SerializedName("familyNameStandardized") val lastNameStandardized: String,
-    @SerializedName("givenNameStandardized") val firstNameStandardized: String?
+    @JsonProperty("dateOfBirth") val dateOfBirthFormatted: String,
+    @JsonProperty("familyNameStandardized") val lastNameStandardized: String,
+    @JsonProperty("givenNameStandardized") val firstNameStandardized: String? = null,
 ) {
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```

Comment:
Better make it lazy or a computed property. Could be quite heavy on instantiation

Reply:
updated